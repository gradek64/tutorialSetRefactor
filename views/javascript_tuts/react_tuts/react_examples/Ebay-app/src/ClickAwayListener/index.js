import React from 'react';
import PropTypes from 'prop-types';

const style = {
  position: 'absolute',
  top: '0',
  left: '0',
  width: '100%',
  height: '100vh',
};
/* handleKeyPress is required by airbnb standard lint but in this case is redundant
cause div is not focusable so it wont work  */
const handleKeyPress = (e) => {
};
const ClickAwayListenerComp = (styles) => {
  const ClickAwayListener = (props) => {
    const { clickAway } = props;
    return (
      <div
        style={styles}
        className="pageMask"
        role="presentation"
        onClick={() => { clickAway(true); }}
        onKeyPress={handleKeyPress}
      />
    );
  };

  ClickAwayListener.propTypes = {
    clickAway: PropTypes.instanceOf(Function).isRequired,
  };

  return ClickAwayListener;
};

export default ClickAwayListenerComp(style);
