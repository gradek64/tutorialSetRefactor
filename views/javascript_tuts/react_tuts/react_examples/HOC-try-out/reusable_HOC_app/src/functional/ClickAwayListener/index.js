import React from "react";

const style = {
  position: "absolute",
  top: "0",
  left: "0",
  width: "100%",
  height: "100vh"
};
const ClickAwayListener = styles => props => (
  <div
    style={styles}
    onClick={() => {
      props.clickAway(true);
    }}
  />
);

export default ClickAwayListener(style);
