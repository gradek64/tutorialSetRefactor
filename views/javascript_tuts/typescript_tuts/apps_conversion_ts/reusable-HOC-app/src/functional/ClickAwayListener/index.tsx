
//typescript cant not find defualt export on React module there needs expoerting * named as React 
import * as React from 'react';
//drop that in tsconfig to allow it : "allowSyntheticDefaultImports": true
import  * as PropTypes from 'prop-types';
//u need to install modules for css types a bit redundant but needed for inline styles

const ClickAwayListenerStyle: React.CSSProperties = {
  position: 'absolute',
  top: '0',
  left: '0',
  width: '100%',
  height: '100vh',
};
/* handleKeyPress is required by airbnb standard lint but in this case is redundant
cause div is not focusable so it wont work  */
const handleKeyPress = (e:React.KeyboardEvent<HTMLDivElement>) => e
const ClickAwayListenerComp = (styles:React.CSSProperties) => {
  //declare your functional component ClickAwayListener with prop clickAway of type function
  const ClickAwayListener:React.FC<{clickAway:(toggleVisible:Boolean)=>void}> = ({clickAway}) => {
    //const { clickAway } = props; this redundant cause it could be destructured as above
    return (
      <div
        style={styles}
        className="pageMask"
        role="presentation"
        onClick={() => { clickAway(true); }}
        onKeyPress={handleKeyPress}
      />
    );
  };
  //this is also redundat since typescript knows that clickAway function is required by default
  /* ClickAwayListener.propTypes = {
    clickAway: PropTypes.instanceOf(Function).isRequired,
  }; */

  return ClickAwayListener;
};

export default ClickAwayListenerComp(ClickAwayListenerStyle);
