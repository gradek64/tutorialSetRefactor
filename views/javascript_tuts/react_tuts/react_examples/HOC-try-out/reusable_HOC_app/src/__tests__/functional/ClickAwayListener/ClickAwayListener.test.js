import React from 'react';
import { shallow } from 'enzyme';

import ClickAwayListener from '../../../functional/ClickAwayListener';

test('renders correctly', () => {
  const wrapper = shallow(<ClickAwayListener clickAway={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

const promise = () => new Promise((resolve) => { setTimeout(() => { resolve('done'); }, 1000); });
test('disapear', async () => {
  const callback = jest.fn();
  const wrapper = shallow(<ClickAwayListener clickAway={() => { callback(true); }} />);

  wrapper.find('.pageMask').simulate('click');
  /*
    *@Idealy U want to test if <div class=pageMask /> after click
    *@disapears but it dooent happen in this component so not likely testable
  */
  await promise().then((res) => {
    console.log('promise resolved');
  });
  // you can click if you clickAway prop method has been called after div click;
  expect(callback).toHaveBeenCalledWith(true);
});
