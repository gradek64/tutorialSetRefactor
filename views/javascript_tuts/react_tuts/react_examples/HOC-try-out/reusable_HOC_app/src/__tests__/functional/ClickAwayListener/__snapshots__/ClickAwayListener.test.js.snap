// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="pageMask"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="presentation"
  style={
    Object {
      "height": "100vh",
      "left": "0",
      "position": "absolute",
      "top": "0",
      "width": "100%",
    }
  }
/>
`;
